javascript/javascript1/week2/smart-ease.io.js

```diff
@@ -0,0 +1,189 @@
+    return "Jacket and pants";
+  } else if (temp < 10) {
+    return "a parka, biker jacket or leather jacket";
+  } else if (temp < 20) {
+    return "Hoodie and jeans";
+  } else if (temp < 30) {
+    return "T-shirt and shorts";
+  } else if (temp < 50) {
+    return "thawb and kufiyah";
+  } else {
+    return "spacesuite";
+  }
+}
+
+alert(
+  `You probably will be better of wearing ${shouldWear(
+    Number(prompt("Enter current tempreture in C:"))
+  )}`
+);
+
+//Fifth task code: Student manager
+const class23Students = [];
+function addStudentToClass(studentName) {
+  if (class23Students.indexOf(studentName) !== -1) {
+    console.log(`Student ${studentName} is already in class23`);
+  } else if (studentName == "") {
+    console.log("The student's name can't be an empty string!");
+  } else if (class23Students.length < 6 || studentName === "Queen") {
+    class23Students.push(studentName);
+  } else {
+    console.log("Class 23 is full");
+  }
+}
+
+function getNumberOfStudents() {
+  return class23Students.length;
+}
+
+for (let i = 0; i < 4; i++) {
+  addStudentToClass(`Student${i}`);
+}
+addStudentToClass("Student1");
+addStudentToClass("");
+addStudentToClass("Anakin Skywalker");
+addStudentToClass("Darth Wader");
+addStudentToClass("Queen");
+addStudentToClass("Princess Amidala");
+
+console.log(getNumberOfStudents());
+console.log(class23Students);
+
+//Sixth, bonus, task code: Candy helper
+const candyPrice = {
+  Sweet: 0.5,
+  Chocolate: 0.7,
+  Toffee: 1.1,
+  "Chewing-gum": 0.03,
+};
+const boughtCandyPrices = [];
+const amountToSpend = Math.floor(Math.random() * 100);
+console.log(`You can buy candies for ${amountToSpend} credits`);
+
+function addCandy(candyType, weight) {
+  boughtCandyPrices.push(Number((candyPrice[candyType] * weight).toFixed(3)));
+  console.log(
+    `Your order is: ${boughtCandyPrices}. The last addition is ${weight} grams of ${candyType} for ${
+      candyPrice[candyType]
+    } per gram, in total we added: ${
+      candyPrice[candyType] * weight
+    } to your order`
+  );
+}
+
+function canBuyMoreCandy(order, allowance) {
+  if (getTotalPrice(order) < allowance) {
+    console.log("You can buy more, so please do!");
+    return true;
+  } else {
+    console.log("Enough candy for you!");
+    return false;
+  }
+}
+
+function getTotalPrice(order) {
+  //did not understood, why the assignment asks to use while loop in stead of "for .. of .." loop
+  let i = 0;
+  let total = 0;
+  while (order[i]) {
+    total += order[i];
+    i++;
+    // Alternatively could have written: if (order.reduce((sum, current) => sum + current, 0) < allowance) { ... }
+    // this probably would work faster then the function I wrote...
+  }
+  console.log(`You already have ordered for ${total} credits`);
+  return total;
+}
+
+function getRandomProperty(obj) {
+  const keys = Object.keys(obj);
+
+  return keys[Math.floor(Math.random() * keys.length)];
+}
+
+while (canBuyMoreCandy(boughtCandyPrices, amountToSpend)) {
```

Comment:
I don't inderstand what is going on here :).

Reply:
It's that part of the code, which fills the array with bought sweets by randomly choosing them from the object and by assigning random quantities. It has nothing to do with the task though. But it's me being lazy: I just didn't want to fill many different types of sweets with different quantities by hand when I was testing the logic of other functions.